### ISSUE TITLE
Can't run SAM example successfully on Safari

### ISSUE THREAD

**Issue by lichin-lin** [huggingface/candle#1162]
#### Description:
SAM model can't load successfully in the Safari browser. I've tried the other example (Whisper, YOLO, Phi), same error happened as well. Is there a chance that Candle can't run in Safari env?

![Screenshot 2023-10-23 at 7 34 36 PM](https://github.com/huggingface/candle/assets/10290616/ceb82e96-66b3-4ff7-b6bc-b32e0b29d841)

#### Environment
Mac M2 / Safari Version 16.5.2 (18615.2.9.11.10)

#### Error messages:
```
Fetch API cannot load https://hf.co/api/s3proxy?GET=https%3A%2F%2Fs3.us-east-1.amazonaws.com%2Flfs.huggingface.co%2Frepos%2Fab%2F9a%2Fab9a34974b39ceddd110b12cedfdc0538851db73ce8ef8482354148ef28f8616%2Fe1b97ef5bb8d7dda5f0d1475e02b666119e883fec44bfaaac25c2e985c7d0af0%3FX-Amz-Algorithm%3DAWS4-HMAC-SHA256%26X-Amz-Content-Sha256%3DUNSIGNED-PAYLOAD%26X-Amz-Credential%3DAKIA4N7VTDGO27GPWFUO%252F20231023%252Fus-east-1%252Fs3%252Faws4_request%26X-Amz-Date%3D20231023T183333Z%26X-Amz-Expires%3D259200%26X-Amz-Signature%3D0b0d5096f1c927cf018216e287efff228db7ad459e059c1d534e6b844177bdc5%26X-Amz-SignedHeaders%3Dhost%26response-content-disposition%3Dattachment%253B%2520filename%252A%253DUTF-8%2527%2527m_bg.wasm%253B%2520filename%253D%2522m_bg.wasm%2522%253B%26response-content-type%3Dapplication%252Fwasm%26x-id%3DGetObject&HEAD=https%3A%2F%2Fs3.us-east-1.amazonaws.com%2Flfs.huggingface.co%2Frepos%2Fab%2F9a%2Fab9a34974b39ceddd110b12cedfdc0538851db73ce8ef8482354148ef28f8616%2Fe1b97ef5bb8d7dda5f0d1475e02b666119e883fec44bfaaac25c2e985c7d0af0%3FX-Amz-Algorithm%3DAWS4-HMAC-SHA256%26X-Amz-Content-Sha256%3DUNSIGNED-PAYLOAD%26X-Amz-Credential%3DAKIA4N7VTDGO27GPWFUO%252F20231023%252Fus-east-1%252Fs3%252Faws4_request%26X-Amz-Date%3D20231023T183333Z%26X-Amz-Expires%3D259200%26X-Amz-Signature%3Db082227f29c0e07aefc4d7f0fa268ca2acf015c1e41025a1e64c3fad6bab2342%26X-Amz-SignedHeaders%3Dhost&sign=eyJhbGciOiJIUzI1NiJ9.eyJyZWRpcmVjdF9kb21haW4iOiJzMy51cy1lYXN0LTEuYW1hem9uYXdzLmNvbSIsImlhdCI6MTY5ODA4NjAxMywiZXhwIjoxNjk4MTcyNDEzLCJpc3MiOiJodHRwczovL2h1Z2dpbmdmYWNlLmNvIn0.i6Dw5p0IdIORGDJmIV6yysyNcNX3Cq2d3kRh6Xl9MqE due to access control checks.
```

```
Unhandled Promise Rejection: DataCloneError: The object can not be cloned.
```

#### Demo video:
https://github.com/huggingface/candle/assets/10290616/b721c1f2-2e63-4ec6-b01a-718d92bc9dcd

**Comment by LaurentMazare** [comment]
Tagging @radames as the expert on these UIs.

**Comment by radames** [comment]
hi @lichin-lin, I'm looking into this now, but the main issue is our hub serving the model and Safari doesn't like the redirects resolving the model file. I'll ping our infra team. Meanwhile one solution is to serve the model file from the same server your hosting your SAM Wasm app.

**Comment by radames** [comment]
hi @lichin-lin, infra team solve the internal issue, it was very specific to the demo hosted on Spaces, I hope you didn't experience issue on your Figma plugin

**Comment by lichin-lin** [comment]
@radames it's working now! thank you, @LaurentMazare, and the infra team! I think I can close the issue :)

I hosted the model file by myself for the plugin, so I didn't encounter the issue 🙌, however, I got another issue for my Figma plugin when running on Safari (that should be something to do with Figma plugin system) - the blob file can't load in.

![Screenshot 2023-10-23 at 11 59 48 PM](https://github.com/huggingface/candle/assets/10290616/76ea6908-0eaa-40aa-aa1b-ec8bd3f0de8b)
